impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -192,10 +192,17 @@ public void start() {
                 .pollInterval(Duration.ofMillis(50))
                 .until(() -> {
                     // Hopefully we can remove this once a fix for https://issues.apache.org/jira/browse/KAFKA-14908 actually lands.
-                    server.startup();
-                    return true;
+                    try {
+                        server.startup();
+                        return true;
+                    }
+                    catch (KafkaException e) {
+                        LOGGER.log(System.Logger.Level.WARNING, "failed to start server due to: " + e.getMessage(), e);
+                        server.awaitShutdown();
```

Comment:
Are you sure about that in the KRaft case? It looks to me like if the broker role brokers were to fail only those would shutdown.  I think calling `server.shutdown()` in your catch block would do no harm and keep the code easier to reason about.  

As the catch block in `kafka/server/KafkaServer.scala:565` catches, rethrows `Thowable`, what would you think about broadening your catch block to catch `Throwable`?

Reply:
I was calling `server.shutdown` in one version (locally) and took it out as superfluous but given you've had the same thoughts I'll add it back.

Reply:
Re throwable. I've never liked catching throwable as I don't believe we should catch error as there is nothing sensible we can do to recover. 

Having said that this is a test framework we could match Kafka and do so.

Reply:
fixed both in b759c0f7fa69da4a36c44c6160fb8d7c824e8e44

Reply:
Hmm with shutdown in place starting to see
```
java.lang.NullPointerException: Cannot invoke "org.apache.kafka.common.metrics.Metrics.sensor(String)" because "metrics" is null
	at kafka.network.SocketServer.<init>(SocketServer.scala:91) ~[kafka_2.13-3.4.0.jar:?]
	at kafka.server.ControllerServer.startup(ControllerServer.scala:179) ~[kafka_2.13-3.4.0.jar:?]
	at kafka.server.KafkaRaftServer.$anonfun$startup$1(KafkaRaftServer.scala:98) ~[kafka_2.13-3.4.0.jar:?]
	at kafka.server.KafkaRaftServer.$anonfun$startup$1$adapted(KafkaRaftServer.scala:98) ~[kafka_2.13-3.4.0.jar:?]
	at scala.Option.foreach(Option.scala:437) ~[scala-library-2.13.10.jar:?]
	at kafka.server.KafkaRaftServer.startup(KafkaRaftServer.scala:98) ~[kafka_2.13-3.4.0.jar:?]
	at io.kroxylicious.testing.kafka.invm.InVMKafkaCluster.lambda$start$1(InVMKafkaCluster.java:196) ~[classes/:?]
	at org.awaitility.core.CallableCondition$ConditionEvaluationWrapper.eval(CallableCondition.java:99) [awaitility-4.2.0.jar:?]
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:248) [awaitility-4.2.0.jar:?]
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:235) [awaitility-4.2.0.jar:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:317) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) [?:?]
	at java.lang.Thread.run(Thread.java:1589) [?:?]
	```
Which suggests restarting is not actually viable.

Reply:
That might suggest a kafka lifecycling defect in KRaft.  The Kafka API suggests that start/stop/restart should be a supporrted use-case.  @showuon WDYT?